### PR TITLE
Test new SpeechSynthesisUtterance(null/undefined)

### PR DESCRIPTION
**Pull request by foolip**
Spec: https://w3c.github.io/speech-api/#tts-section

This is targeting a mundane difference in `new
SpeechSynthesisUtterance(undefined)` depending on whether the IDL definition of
the construtor uses a single constructor with an optional argument, or two
constructor overloads with zero and one argument respectively.

This is not equivalent because
https://heycam.github.io/webidl/#dfn-overload-resolution-algorithm first
considers the number of arguments, so given one argument the
`Constructor(DOMString text)` overload is used. With optional arguments,
however, trailing undefined arguments are ignored so that `foo(undefined)` will
be indistinguishable from `foo()`.

In other words, `new SpeechSynthesisUtterance(undefined)` should result in a
`text` of "undefined" because the spec uses two contructor overloads, but
Blink/WebKit have a single constructor. Edge and Firefox pass the added test, so
align with them rather than changing the spec. (It doesn't really matter.)

### CODE REVIEW

**Comment by saschanaz** [comment]
Personally an IDL attribute that returns `undefined` is a bit weird, and AFAIK the Web IDL spec cannot represent that behavior (https://github.com/heycam/webidl/issues/60).

And it seems the attribute is [not marked as nullable](https://w3c.github.io/speech-api/#utterance-attributes), maybe a spec bug?

**Comment by foolip** [comment]
It doesn't return undefined, it's the string "undefined" (and "null"). That's because the argument to the constructor is of type DOMString and is thus converted to a string.

**Comment by saschanaz** [comment]
Oh, that's right. Thank you for making things clear.

But can we choose the shorter one (the optional argument one) instead of current overloads, when browsers should anyway change?

**Comment by foolip** [comment]
Yep, it doesn't really matter which form we pick, but I thought that since Edge and Gecko have already matched the spec, it's more fair to leave that be. I need to make a change around the same place in the Chromium code anyway, which is how I noticed this.

**Comment by foolip** [comment]
Although, there is one additional consideration. If we'd ever want to add a  dictionary and have that as an optional argument for the constructor, then we'd probably want to change to a single `Constructor(optional (DOMString or SpeechSynthesisUtteranceInit) init)` overload, in which case the case tested in this PR would change.

Thoughts?

**Comment by saschanaz** [comment]
So one more +1 for optional argument 😄 to ease future changes.

**Comment by foolip** [comment]
@saschanaz can you review this PR as well? Now updated to match https://github.com/w3c/speech-api/pull/48.

**Review by saschanaz — APPROVED** [review]
